### ISSUE TITLE
Documentation on protocol should mark all implementations as documented

### ISSUE THREAD

**Issue by lightsprint09** [realm/SwiftLint#836]
When I have a protocol called AProtocol:

``` swift
protocol AProtocol {
/**
     Foo
     */
    func foo()
}
```

and an imlelemtation of AProtocol:

``` swift
struct AImpl: AProtocol {
    func foo() {
    ///...
    }
}
```

AImpl.foo gets marked as missing documentation. This should not happen. You should not need to duplicate the documentation

**Comment by jpsim** [comment]
This would be great. Although pretty hard to do without making some serious changes to how SwiftLint works.

By adding a mode to SwiftLint that requires that the module 1) already be compiled; and 2) SwiftLint be passed the compiler flags; we could enable a whole new class of rules that can work across files and even modules, allowing to run rules that do things more like a static analyzer would.

**Comment by marcelofabri** [comment]
Closing this since we removed this rule.
